application/src/main/java/org/thingsboard/server/service/subscription/TbSubscription.java

```diff
@@ -49,7 +52,10 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(sessionId, subscriptionId, tenantId, entityId, type);
+        if (hash == 0) {
```

Comment:
Why not use `@EqualsAndHashCode(cacheStrategy = CacheStrategy.LAZY)` (with added exclusions for serviceId, etc.)? Same for UUIDBased

Reply:
the implementation of the Lombok hash cache is quite diff. I did that particular change to maintain full compatibility with previous version